pkg/controller/atlasproject/teams.go

```diff
@@ -36,7 +36,7 @@ func (r *AtlasProjectReconciler) ensureAssignedTeams(workflowCtx *workflow.Conte
 		}
 
 		team := &akov2.AtlasTeam{}
-		teamReconciler := r.teamReconcile(team, project.ConnectionSecretObjectKey())
+		teamReconciler := r.teamReconcile(team, project.ConnectionSecretObjectKey(), project.ID())
```

Comment:
Remove unused arg also in invocation.

```suggestion
		teamReconciler := r.teamReconcile(team, project.ConnectionSecretObjectKey())
```